### ISSUE TITLE
auto and printf : segmentation fault

### ISSUE THREAD

**Issue by TheFausap** [Spydr06/BCause#2]
Hello,
I cannot understand why this simple program generates a segmentation fault:

```
main() {
	extrn printf,ms;
	auto i;

	i = 56;
	printf("%d*n",i);
}
ms 4000;
```

If I change printf with putchar (I know the output will be different), I have no segfault.

```
main() {
	extrn putchar,ms;
	auto i;

	i = 56;
	putchar(i);
}
ms 4000;
```

Any ideas why?

thanks
Fausto

**Comment by Spydr06** [comment]
Your program seems correct, it's probably a compiler bug.
I'll look into it when I have time

**Comment by Spydr06** [comment]
Issue was due to a misaligned stack. Fixed in [adf8d94](https://github.com/Spydr06/BCause/commit/adf8d941b605f7abb8c669d6bd389f6ed9c09953)
